icu4c/source/test/intltest/dtfmttst.cpp

```diff
@@ -5555,6 +5556,51 @@ void DateFormatTest::TestAdoptCalendarLeak() {
     sdf.adoptCalendar(Calendar::createInstance(status));
 }
 
+void DateFormatTest::Test20741_ABFields() {
+    IcuTestErrorCode status(*this, "Test20741_ABFields");
+
+    const char16_t timeZone[] = u"PST8PDT";
+
+    UnicodeString skeleton = u"EEEEEBBBBB"; 
+    Locale locale("en-001");
+    LocalPointer<DateTimePatternGenerator> gen(DateTimePatternGenerator::createInstance(locale, status));
+    UnicodeString pattern = gen->getBestPattern(skeleton, status);
+
+    assertEquals("Pattern", pattern, u"ccccc, BBBBB");
+
+    SimpleDateFormat dateFormat(pattern, locale, status);
+    FieldPositionIterator fpositer;
+    UnicodeString result;
+    LocalPointer<Calendar> calendar(Calendar::createInstance(TimeZone::createTimeZone(timeZone), status));
+    calendar->setTime(UDate(0), status);
+    dateFormat.format(*calendar, result, &fpositer, status);
+
+    assertEquals("Result", result, u"W, in the afternoon");
+
+    // Now we test C API comparing result with C++
+
+    LocalUDateFormatPointer df(udat_open(UDAT_PATTERN, UDAT_PATTERN, locale.getName(),
+        timeZone, std::char_traits<char16_t>::length(timeZone),
+        pattern.getTerminatedBuffer(), pattern.length(), status));
+
+    UChar res[100];
+
```

Comment:
Nit: The grouping of lines is strange.  Use double-space to separate different distinct sections, and single-space within a section.